### DISCUSSION TITLE
Window size limit

### DISCUSSION THREAD

**Discussion by unayakani** [nvim-tree/nvim-tree.lua#1776] [Q&A]
How to make the nvim tree window not exceed a certain size limit?
For example, “In any case, nvim-tree’s size  should be less than or equal to 30”
Thankyou

**Comment by alex-courtis** [comment]
That functionality does not exist. The closest we have is setting the width to a percentage e.g. 

```lua
    view = {
      width = "20%",
```

Please raise a [feature request](https://github.com/nvim-tree/nvim-tree.lua/issues/new/choose)

**Reply by unayakani** [reply]
Good to know!
Btw, how about a startup directory?
I want nivm-tree to open every time at a specific directory

**Reply by unayakani** [reply]
Oh but btw, it is possible with
```vim
        view = {
            adaptive_size = false,
            width = 30,
        }
```

**Reply by alex-courtis** [reply]
> Btw, how about a startup directory?
> I want nivm-tree to open every time at a specific directory

You could do something like `:NvimTreeOpen /path/to/my/dir` or `api.tree.open("/path/to/my/dir")` after invoking nvim-tree setup.

**Reply by alex-courtis** [reply]
> Oh but btw, it is possible with
> 
> ```viml
>         view = {
>             adaptive_size = false,
>             width = 30,
>         }
> ```

`adaptive_size` will override width and will use sizes greater than 30.

**Reply by unayakani** [reply]
> > Oh but btw, it is possible with
> > ```viml
> >         view = {
> >             adaptive_size = false,
> >             width = 30,
> >         }
> > ```
> 
> `adaptive_size` will override width and will use sizes greater than 30.

Adaptive size is `false`

**Reply by unayakani** [reply]
> > Btw, how about a startup directory?
> > I want nivm-tree to open every time at a specific directory
> 
> You could do something like `:NvimTreeOpen /path/to/my/dir` or `api.tree.open("/path/to/my/dir")` after invoking nvim-tree setup.

Elaborate if you can please?

**Reply by alex-courtis** [reply]
> Adaptive size is `false`

My mistake; please accept my apologies.

What is the desired functionality here? My understanding is "window width is variable (adaptive_size) however capped at a specified limit". That is useful, hence I would gratefully appreciate a feature request.

**Reply by alex-courtis** [reply]
> Elaborate if you can please?

`/tmp/nvt-min.lua`
```lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvt-min/site]]
local package_root = "/tmp/nvt-min/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup {
    {
      "wbthomason/packer.nvim",
      "nvim-tree/nvim-tree.lua",
      "nvim-tree/nvim-web-devicons",
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  }
end
if vim.fn.isdirectory(install_path) == 0 then
  print "Installing nvim-tree and dependencies."
  vim.fn.system { "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path }
end
load_plugins()
require("packer").sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua setup()]]
vim.opt.termguicolors = true
vim.opt.cursorline = true

-- MODIFY NVIM-TREE SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
_G.setup = function()
  require("nvim-tree").setup {}
  require("nvim-tree.api").tree.open("/tmp")
end
```

```sh
cd $HOME
nvim -nu /tmp/nvt-min.lua
```

**Reply by unayakani** [reply]
> > Adaptive size is `false`
> 
> My mistake; please accept my apologies.
> 
> What is the desired functionality here? My understanding is "window width is variable (adaptive_size) however capped at a specified limit". That is useful, hence I would gratefully appreciate a feature request.

No problem.
What it does is "Keep the window size a constant. i.e.: 30 and don't change the size at all"

**Reply by unayakani** [reply]
> I would gratefully appreciate a feature request.

Oh my! Are you one of the creators?
I'll see when I'm free to raise a feature request.

**Comment by alex-courtis** [comment]
Please see #1797
